receiver/azureblobreceiver/receiver.go

```diff
@@ -18,22 +18,108 @@ import (
 	"context"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/obsreport"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
 )
 
+type LogsDataConsumer interface {
+	ConsumeLogsJSON(ctx context.Context, json []byte) error
+	SetNextLogsConsumer(nextLogsConsumer consumer.Logs)
+}
+
+type TracesDataConsumer interface {
+	ConsumeTracesJSON(ctx context.Context, json []byte) error
+	SetNextTracesConsumer(nextracesConsumer consumer.Traces)
+}
+
 type blobReceiver struct {
+	blobEventHandler   BlobEventHandler
+	logger             *zap.Logger
+	logsUnmarshaler    plog.Unmarshaler
+	tracesUnmarshaler  ptrace.Unmarshaler
+	nextLogsConsumer   consumer.Logs
+	nextTracesConsumer consumer.Traces
+	obsrecv            *obsreport.Receiver
 }
 
 func (b *blobReceiver) Start(ctx context.Context, host component.Host) error {
-	return nil
+
+	b.blobEventHandler.SetLogsDataConsumer(b)
+	b.blobEventHandler.SetTracesDataConsumer(b)
+
+	err := b.blobEventHandler.Run(ctx)
+
+	return err
 }
 
 func (b *blobReceiver) Shutdown(ctx context.Context) error {
+	b.blobEventHandler.Close(ctx)
+
 	return nil
 }
+func (b *blobReceiver) SetNextLogsConsumer(nextLogsConsumer consumer.Logs) {
+	b.nextLogsConsumer = nextLogsConsumer
+}
+
+func (b *blobReceiver) SetNextTracesConsumer(nextTracesConsumer consumer.Traces) {
+	b.nextTracesConsumer = nextTracesConsumer
+}
+
+func (b *blobReceiver) ConsumeLogsJSON(ctx context.Context, json []byte) error {
+
+	if b.nextLogsConsumer == nil {
+		return nil
+	}
+
+	logsContext := b.obsrecv.StartLogsOp(ctx)
+
+	logs, err := b.logsUnmarshaler.UnmarshalLogs(json)
+	if err == nil {
+		err = b.nextLogsConsumer.ConsumeLogs(logsContext, logs)
+	} else {
+		b.logger.Error(err.Error())
+	}
```

Comment:
Instead of logging the error, just return it and let the caller handle it

```suggestion
	logs, err := b.logsUnmarshaler.UnmarshalLogs(json)
	if err != nil {
		return fmt.Errorf("failed to unmarshal logs: %w", err)
	}
	
	err = b.nextLogsConsumer.ConsumeLogs(logsContext, logs)
```